src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            addFirst(value);
+        } else {
+            addLast(value);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, size + 1);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (tail == null && head == null) {
+            addFirst(value);
+        } else if (index == 0) {
+            Node<T> node = new Node<>(null, value, head);
+            head.prev = node;
+            head = node;
+        } else {
+            Node<T> prev = getNodeByIndex(index - 1);
+            Node<T> node = new Node<>(prev, value, prev.next);
+            prev.next.prev = node;
+            prev.next = node;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index, size);
+        Node<T> node = getNodeByIndex(index);
+        return node.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index, size);
+        Node<T> node = getNodeByIndex(index);
+        T result = node.value;
+        node.value = value;
+        return result;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index, size);
+        Node<T> node = getNodeByIndex(index);
+        unlink(node);
+        size--;
+        return node.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = head;
+        for (int i = 0; i < size; i++) {
+            if ((object == node.value) || (object != null && object.equals(node.value))) {
```

Comment:
```suggestion
            if (object == node.value || object != null && object.equals(node.value)) {
```
redundant brackets (read about operators' priority)